### ISSUE TITLE
New ink! contracts generated/compiled to Wasm after v3.2.0 not accepted in contracts-ui.

### ISSUE THREAD

**Issue by Panasthetik** [paritytech/ink#1282]
Rust stable version:   "1.61.0"
Rust nightly version:  "1.63.0-nightly-x86_64-apple-darwin"
Cargo contract version:  "1.4.0" (latest):

Ever since the ink! dependencies (ink_env, ink_prelude, etc.) )were upgraded to v3.2.0, all new Wasm compiled contracts break in the contracts-ui - both online:

https://paritytech.github.io/contracts-ui/

..And locally on a user's machine.

For example, if you create a new contract:

```cargo contract new yourfilename```

Build the contract correctly with:

```cargo +nightly contract build```

And then:

Upload the contract file bundle in the UI:  ```yourfilename.contract```

You will get THIS error (see bottom right corner)  in the contracts-ui user interface:

![contracterror](https://user-images.githubusercontent.com/88591701/172465027-548f3d32-c47b-4dc0-a639-c18e35f817f0.png)

This problem was not present two weeks ago - I have tried to downgrade cargo-contract to version 1.3.0, or downgrade dependencies to "3.0.0" in the Cargo.toml file - and it still does not correct this problem, because when it compiles, it still pulls the "v3.2.0" ink! crates when it generates the Wasm blob.

NOTE: I also compared previously-compiled Wasm contract bundles that were built 2-3 weeks ago (when ink! dependencies crates were still at 3.1 or below), and these are OK in the contracts-ui. You can deploy them correctly.

The issue seems to be one of the following:

1) The ink! Rust crates for v3.2.0, when compiled to Wasm, result in a slightly different Wasm blob that is not recognized by the contracts-ui correctly.

and/or:

2) The contracts-ui application needs to be fixed/updated to accommodate any changes to the accepted code output (Wasm).

Until then, any new contracts generated with:

```cargo contract new```

May not be deployable in the contracts-ui.

**Comment by statictype** [comment]
please make sure you are using the latest version of [substrate-contracts-node](https://github.com/paritytech/substrate-contracts-node)

**Comment by Panasthetik** [comment]
Thanks for your timely response! - I have downloaded the latest ```substrate-contracts-node``` and recompiled the Substrate Contracts Node:

```cargo build --release```

I have started up the node on my machine with:

```./substrate-contracts-node --dev``` 

A new contract generated with:

```cargo contract new testingcontract```
```cargo +nightly contract build```

new file ```testingcontract.contract``` uploaded to:

https://paritytech.github.io/contracts-ui/

Everything is now fine -  and all contracts now load correctly with ink! v3.2.0:

![paritytest](https://user-images.githubusercontent.com/88591701/172684335-7495b51b-9110-40a9-ae4c-ff0d86a40773.png)

I appreciate your help and am now closing this issue.

Best Regards
